src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareIncludeExcludeNodeOperation.cs

```diff
@@ -58,7 +65,45 @@ public void Execute(TaskExecutionMode mode)
                     throw new InvalidOperationException(SR.SchemaCompareExcludeIncludeNodeNotFound);
                 }
 
+                // Check first if the dependencies will allow this if it's an exclude request
+                if (!this.Parameters.IncludeRequest)
+                {
+                    IEnumerable<SchemaDifference> dependencies = this.ComparisonResult.GetExcludeDependencies(node);
+
+                    bool block = false;
+                    foreach (SchemaDifference entry in dependencies)
+                    {
+                        if (entry.Included)
+                        {
+                            block = true;
+                            break;
+                        }
+                    }
+
+                    if (block)
+                    {
+                        this.Success = false;
+                        return;
+                    }
+                }
+
                 this.Success = this.Parameters.IncludeRequest ? this.ComparisonResult.Include(node) : this.ComparisonResult.Exclude(node);
+
+                // create list of affected dependencies of this request
+                if (this.Success)
+                {
+                    IEnumerable<SchemaDifference> dependencies = this.ComparisonResult.GetIncludeDependencies(node);
+
+                    this.ChangedDifferences = new List<DiffEntry>();
+                    if (dependencies != null)
+                    {
+                        foreach (SchemaDifference difference in dependencies)
+                        {
+                            DiffEntry diffEntry = SchemaCompareUtils.CreateDiffEntry(difference, null);
+                            this.ChangedDifferences.Add(diffEntry);
+                        }
```

Comment:
Could you linqify this?
this.ChangedDifferences = dependencies.Select(d => SchemaCompareUtils.CreateDiffEntry(difference, null)).ToList()